### PR TITLE
Elaborate name matching in user guide

### LINKED ISSUE #14060: Extend name matching section in user guide
The user guide does not mention that the name matching also works for projects. We should also document the new kebab case matching.

### PR DESCRIPTION
**Pull request by donat**
Fixes #14060

The rendered version of the updated section:
image

### CODE REVIEW

**Comment by donat** [comment]
@gradle-bot test this

**Comment by donat** [comment]
@bot-gradle test this

File: subprojects/docs/src/docs/userguide/reference/command_line_interface.adoc [hunk 1]
```diff
@@ -140,12 +140,16 @@ If a task fails, any subsequent tasks that were depending on it will not be exec
 
 When you specify tasks on the command-line, you don’t have to provide the full name of the task. You only need to provide enough of the task name to uniquely identify the task. For example, it's likely `gradle che` is enough for Gradle to identify the `check` task.
 
-You can also abbreviate each word in a camel case task name. For example, you can execute task `compileTest` by running `gradle compTest` or even `gradle cT`.
+The same applies for project names. You can execute the run the `check` task in the `library` subproject with the `gradle lib:che` command.
```
Comment (jjohannes) [inline subprojects/docs/src/docs/userguide/reference/command_line_interface.adoc:143]:
```suggestion
The same applies for project names. You can execute the `check` task in the `library` subproject with the `gradle lib:che` command.
```

File: subprojects/docs/src/docs/userguide/reference/command_line_interface.adoc [hunk 2]
```diff
@@ -140,12 +140,16 @@ If a task fails, any subsequent tasks that were depending on it will not be exec
...
 
-.Abbreviated camel case task name
+You can use https://en.wikipedia.org/wiki/Camel_case[camel case] patterns for more complex abbreviations. These patterns are expanded to match to camel case and https://en.wikipedia.org/wiki/Kebab_case[kebab case] names. For example the pattern `foBa` (or even `fB`) matches to `fooBar` and to `foo-bar`.
```
Comment (jjohannes) [inline subprojects/docs/src/docs/userguide/reference/command_line_interface.adoc:145]:
```suggestion
You can use https://en.wikipedia.org/wiki/Camel_case[camel case] patterns for more complex abbreviations. These patterns are expanded to match camel case and https://en.wikipedia.org/wiki/Kebab_case[kebab case] names. For example the pattern `foBa` (or even `fB`) matches `fooBar` and `foo-bar`.
```

File: subprojects/docs/src/docs/userguide/reference/command_line_interface.adoc [hunk 3]
```diff
@@ -140,12 +140,16 @@ If a task fails, any subsequent tasks that were depending on it will not be exec
...
+
+More concretely, you can run the `compileTest` task in the `my-awesome-library` subproject with the `gradle  mAL:cT` command.
```
Comment (jjohannes) [inline subprojects/docs/src/docs/userguide/reference/command_line_interface.adoc:147]: Would be nice to use a task name that exists for real in many builds as example - like `compileTestJava` (just a suggestion).

**Pull request merged**